src/node.cc

```diff
@@ -2867,9 +2869,54 @@ static bool ExecuteBootstrapper(Environment* env, Local<Function> bootstrapper,
 }
 
 
+void PromiseRejectCallback(PromiseRejectMessage message) {
+  Local<Promise> promise = message.GetPromise();
+  Isolate* isolate = promise->GetIsolate();
+  PromiseRejectEvent event = message.GetEvent();
+
+  Environment* env = Environment::GetCurrent(isolate);
+  Local<Function> callback;
+  Local<Value> value;
+
+  if (event == v8::kPromiseRejectWithNoHandler) {
+    callback = env->promise_reject_unhandled_function();
+    value = message.GetValue();
+
+    if (value.IsEmpty())
+      value = Undefined(isolate);
+  } else if (event == v8::kPromiseHandlerAddedAfterReject) {
+    callback = env->promise_reject_handled_function();
+    value = Undefined(isolate);
+  } else {
+    UNREACHABLE();
+  }
+
+  Local<Value> args[] = { promise, value };
+  MaybeLocal<Value> ret = callback->Call(env->context(),
+                                         Undefined(isolate),
+                                         arraysize(args),
+                                         args);
+
+  if (!ret.IsEmpty() && ret.ToLocalChecked()->IsTrue())
+    env->tick_info()->promise_rejections_toggle_on();
+}
+
 void LoadEnvironment(Environment* env) {
+  LoadEnvironmentGeneric(env, true, true);
+}
+
+void LoadEnvironmentGeneric(Environment* env, bool standalone_mode,
+                            bool upstream_node_mode) {
   HandleScope handle_scope(env->isolate());
 
+  if (standalone_mode) {
+    env->isolate()->AddMessageListener(OnMessage);
+    env->isolate()->SetFatalErrorHandler(OnFatalError);
+  }
+  if (upstream_node_mode) {
+    env->isolate()->SetPromiseRejectCallback(PromiseRejectCallback);
+  }
```

Comment:
I still feel like this would be better kept for `NewIsolate()` – if that doesn’t work, can we still leave it out of `LoadEnvironment()` and maybe put it in its own function (e.g. `node::AttachErrorHandlers(Isolate* isolate)`)?